File: src/WorkBC.Data/Model/JobBoard/SystemSetting.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using WorkBC.Data.Enums;

namespace WorkBC.Data.Model.JobBoard
{
    public class SystemSetting
    {
        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        [MaxLength(400)]
        [Key]
        public string Name { get; set; }

        public string Value { get; set; }

        public string DefaultValue { get; set; }

        public string Description { get; set; }

        public SystemSettingFieldType FieldType { get; set; }

        [ForeignKey("AdminUser")]
        public int ModifiedByAdminUserId { get; set; }

        public virtual AdminUser ModifiedByAdminUser { get; set; }

        public DateTime DateUpdated { get; set; }
    }
}

File: jobboard-laravel/app/Models/SystemSetting.php
<?php

namespace App\Models;

use App\Models\Casts\TolerantEnum;
use App\Models\Enums\SystemSettingFieldType;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

final class SystemSetting extends Model
{
    protected $table = 'SystemSettings';

    protected $primaryKey = 'Name';

    protected $keyType = 'string';

    public $incrementing = false;

    public $timestamps = false;

    protected $casts = [
        'FieldType' => TolerantEnum::class.':'.SystemSettingFieldType::class,
        'DateUpdated' => 'datetime',
    ];

    public function modifiedByAdminUser(): BelongsTo
    {
        return $this->belongsTo(AdminUser::class, 'ModifiedByAdminUserId', 'Id');
    }
}
